### PR TITLE
Documentation for Extraction Engines

### LINKED ISSUE #103: Add docs for extraction engines
Add docs for the following extraction engines:

- `CsvExtractionEngine`
- `JsonExtractionEngine`

### PR DESCRIPTION
**Pull request by trish11953**
Resolves #103 

----
:books: Documentation preview :books:: https://griptape--130.org.readthedocs.build/

### CODE REVIEW

File: docs/griptape-framework/data/extraction-engines.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
+# Overview 
+Extraction Engines are used to extract data from text.
```
Comment (collindutter) [inline docs/griptape-framework/data/extraction-engines.md:2]: Let's add another sentence or two here. Maybe describe some use cases or types of extraction.

File: docs/griptape-framework/data/extraction-engines.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
+# Overview 
+Extraction Engines are used to extract data from text. 
+
+## CSV Extraction Engine
+
+Used to extract data from text using a CSV file. You can set a custom [prompt driver](../../griptape-framework/structures/prompt-drivers.md) or [chunker](../../griptape-framework/data/chunkers.md).
```
Comment (collindutter) [inline docs/griptape-framework/data/extraction-engines.md:6]: Capitalize `Prompt Driver` and `Chunker`

File: docs/griptape-framework/data/extraction-engines.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+Used to extract data from text using a CSV file. You can set a custom [prompt driver](../../griptape-framework/structures/prompt-drivers.md) or [chunker](../../griptape-framework/data/chunkers.md). 
+You can use it along with [Extraction Tasks](../../griptape-framework/structures/tasks.md)
+
+```python
+from griptape.engines import CsvExtractionEngine
+from griptape.artifacts import ListArtifact, ErrorArtifact
+from griptape.drivers import OpenAiChatPromptDriver
+
+# Initialize the CsvExtractionEngine instance
+csv_engine = CsvExtractionEngine(prompt_driver=OpenAiChatPromptDriver(model="gpt-4", temperature=0.3))
```
Comment (collindutter) [inline docs/griptape-framework/data/extraction-engines.md:15]: What's the reasoning for explicitly providing a Prompt Driver? Does this only work with this temperature?

File: docs/griptape-framework/data/extraction-engines.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+```
+```
+{"id": "c0aa68f91f5e4368b784b7e20c5187bc", "name": "c0aa68f91f5e4368b784b7e20c5187bc", "type": "CsvRowArtifact", "value": {"name": "Alice", "age": "\"28\"", "location": "\"New York\""}}
+{"id": "1e02e3609eb3443cbe5072830515999f", "name": "1e02e3609eb3443cbe5072830515999f", "type": "CsvRowArtifact", "value": {"name": "Bob", "age": "\"35\"", "location": "\"California\""}}
+{"id": "a685dc0d9cd24feb9c08f9010dce0387", "name": "a685dc0d9cd24feb9c08f9010dce0387", "type": "CsvRowArtifact", "value": {"name": "Charlie", "age": "\"40\"", "location": "\"Texas\""}}
+```
+
+## JSON Extraction Engine
+
+Used to extract data from text using a JSON file. You can set a custom [prompt driver](../../griptape-framework/structures/prompt-drivers.md) or [chunker](../../griptape-framework/data/chunkers.md).
```
Comment (collindutter) [inline docs/griptape-framework/data/extraction-engines.md:43]: Capitalize `Prompt Driver` and `Chunker`

Comment (collindutter) [inline docs/griptape-framework/data/extraction-engines.md:43]: Can you please explain what you mean by "using a JSON file"?

**Review by collindutter — CHANGES_REQUESTED** [review]

File: docs/griptape-framework/data/extraction-engines.md (hunk 2 above)
Comment (collindutter) [inline docs/griptape-framework/data/extraction-engines.md:6]: Can you please explain what you mean by "using a CSV file"?

**Review by collindutter — CHANGES_REQUESTED** [review]

File: docs/griptape-framework/data/extraction-engines.md (hunk 2 above)
Comment (trish11953) [reply docs/griptape-framework/data/extraction-engines.md:6]: Done!

File: docs/griptape-framework/data/extraction-engines.md (hunk 1 above)
Comment (trish11953) [reply docs/griptape-framework/data/extraction-engines.md:2]: Sure!

File: docs/griptape-framework/data/extraction-engines.md (hunk 4 above)
Comment (trish11953) [reply docs/griptape-framework/data/extraction-engines.md:43]: Done!

Comment (trish11953) [reply docs/griptape-framework/data/extraction-engines.md:43]: it should be "Used to extract data from a JSON file"

File: docs/griptape-framework/data/extraction-engines.md (hunk 2 above)
Comment (trish11953) [reply docs/griptape-framework/data/extraction-engines.md:6]: it should be "Used to extract data from a CSV file"

File: docs/griptape-framework/data/extraction-engines.md (hunk 3 above)
Comment (trish11953) [reply docs/griptape-framework/data/extraction-engines.md:15]: There was no reason to use the prompt driver, just added it to the example so people know that a prompt driver can be used. The temperature was totally random. I can remove the prompt driver from the example if you want.

**Review by collindutter — APPROVED** [review]

**Pull request merged**